pkg/core/resources/manager/cache.go

```diff
@@ -85,7 +85,10 @@ func (c *cachedManager) Get(ctx context.Context, res model.Resource, fs ...store
 
 func (c *cachedManager) List(ctx context.Context, list model.ResourceList, fs ...store.ListOptionsFunc) error {
 	opts := store.NewListOptions(fs...)
-	cacheKey := fmt.Sprintf("LIST:%s:%s", list.GetItemType(), opts.HashCode())
+	if !opts.IsCacheable() {
+		return fmt.Errorf("filter functions are not allowed for cached store")
+	}
+	cacheKey := fmt.Sprintf("LIST:%s:%s:%t:%s:%d:%s", list.GetItemType(), opts.HashCode(), opts.Ordered, opts.NamePrefix, opts.PageSize, opts.PageOffset)
```

Comment:
Probably these `opts.Ordered, opts.NamePrefix, opts.PageSize, opts.PageOffset` should be moved inside `opts.HashCode()`:
```go
func (l *ListOptions) HashCode() string {
	return strings.Join([]string{l.Mesh, l.Ordered, l.NamePrefix, l.PageSize, l.PageOffset}, ":")
}
```

Reply:
Cannot make join because not all types are string, did `fmt.Sprintf`